lib/vehicle_factory.rb

```diff
@@ -0,0 +1,30 @@
+require './lib/dmv_data_service'
+require './lib/vehicle'
+
+class VehicleFactory
+
+    attr_accessor :vehicle_details,
+                  :vehicles
+
+    def initialize
+        @vehicle_details = {vin: nil, year: nil, make: nil, model: nil, engine: :ev, plate_type: nil, registration_date: nil}
```

Comment:
Ok, the reason I'm not liking this attribute in the Factories is because they are not an attribute of the VehicleFactory class, these are attributes of the Vehicles you're creating. I could see an argument to keeping them here in the initialize, but I also think it would be better to keep these local within the method you're writing to reduce unnecessary setting of default values. 
There are also other ways of setting default values so you don't need to explicitly write out all the keys like this. Actuallly, since most of these default values are already nil, we don't need to set the hash at all like this since the defaults will already start as nil.